### DISCUSSION TITLE
Future Development

### DISCUSSION THREAD

**Discussion by jakemegrian** [hwnd-git/Diablo-IV-AHK-WASD#1] [General]
This is an excellent script that works very well! I'm very happy that I am able to play Diablo 4 with WASD and not hurt my hands due to repetitive strain injury from spamming left click to move.

I read through the "Potential Improvements" section and all those feature would be greatly appreciated!

Will you be continuing to develop this script?

**Comment by hwnd-git** [comment]
Hey, I appreciate the kind words, thanks!

As for the improvements.
1. There is a newer, or maybe rather alternative version of the script now (https://github.com/hwnd-git/Diablo-IV-AHK-WASD/blob/main/D4_WASD.ahk) that is less spammy and has the randomization built in.
2. I do not plan to implement movement smoothing in the foreseeable future. Unfortunately the effort of implementing it would outweigh the slight aesthetic gain. Being 100% frank - I'm no-lifing in the game too hard right now. No time for coding ;D
3. I was thinking about the feature where you can define 'kiting' interval. Within each interval, script triggers a skill and then moves the character right away. But I abandoned it for two reasons. First is that if felt too automatic and I'd rather mitigate the risks of getting banned instead of reinforcing them. Secondly, I am pretty happy with how the script works as it is.
